Lesson_22/js/selector.js

```diff
@@ -0,0 +1,60 @@
+export class Selector {
+   #target = null;
+   #lastSelectedIndex = null;
+
+   constructor(list) {
+      this.list = list;
+      this.items = list.children;
+
+      list.addEventListener('click', this.onClick.bind(this));
+      list.addEventListener('mousedown', e => e.preventDefault());
+   }
+
+   #clearSelected() {
+      for (let li of this.items) {
+         li.classList.remove('selected');
+      }
+   }
+
+   #addSelected(target) {
+      target.classList.add('selected');
+   }
+
+   #getIndex(item) {
+      return [...this.items].indexOf(item);
+   }
+
+   onClick(e) {
+      if (e.target.tagName !== "LI") return;
+
+      this.#target = e.target;
+
+      if (e.ctrlKey || e.metaKey) {
+         this.toggleSelected();
+      } else if (e.shiftKey) {
+         this.multipleSelect();
```

Comment:
i'd recommend you to pass 2 arguments in this function (target, lastTarget). It should help